packages/smooth_app/lib/pages/image/product_image_widget.dart

```diff
@@ -57,20 +34,21 @@ class _ProductImageWidgetState extends State<ProductImageWidget> {
     final DateFormat dateFormat =
         DateFormat.yMd(ProductQuery.getLanguage().offTag);
 
-    darkBackground = darkBackground ?? true;
-
     final Widget image = SmoothImage(
-      width: widget.squareSize,
-      height: widget.squareSize,
+      cacheHeight:
```

Comment:
No `cacheWidth`?

It's a bit strange to say "you can force `ImageSize` if you want, but regardless we will force a `cacheHeight` even if you don't want to and there's nothing you can do about it"

Reply:
Here we have a square, but the server's photo may have a different ratio.
If I say, `cacheWidth` == `cacheHeight`, it means the server picture is a square.
That's why, I only provide one of the argument

Reply:
I understand the square effect.

I don't know how relevant this cache option would be, performance-wise.
My assumption is that the best performance improvement is with smaller images (400).

I also fear that this cache option entail side-effects: what if I download a full image with that cache option for a thumbnail, and then display the full image on a full screen after a tap? From the docs, I assume that the cached image will be stored as resized, and then reused, which would render a pixelized image.

At least, you could add a comment about why cacheHeight but not cacheWidth.